direnvrc

```diff
@@ -140,7 +140,7 @@ touch "$PWD/.envrc"
 touch -r "$PWD/.envrc" "${layout_dir}"/*.rc
 EOF
 
-  if [[ ! -x "${layout_dir}/bin/nix-direnv-reload" ]]; then
+  if [[ "$(stat --format=%A "${layout_dir}/bin/nix-direnv-reload")" != *x ]]; then
     chmod +x "${layout_dir}/bin/nix-direnv-reload"
   fi
```

Comment:
If we are already calling binaries we might as well can just call `chmod` unconditionally. However I am bit suspicious about the error. "Bad interpreter" seems to actually refer `/usr/bin/env bash` not working than this script not having the executable bit.

Reply:
Ah. I re-read your issue again. `access()` is indeed a different syscall from stat.